### PR TITLE
Allow llvm16

### PR DESCRIPTION
**Pull request by mbouaziz**
Part of my work towards #734

I probably won't go further for the time being.

Before merging, I want to:
- push docker images with LLVM 16 and skiplang built with it
- check all CI jobs are still green

### CODE REVIEW

**Comment by beauby** [comment]
Nice. If we start supporting several versions of `LLVM`, we should have CI jobs for the compiler tests using each supported (major) version.

**Comment by mbouaziz** [comment]
We probably don't want to have the burden of maintaining compatibility with several versions, either they are 99% compatible and we only have to worry when someone enters the 1% edge case, or we have to stick to one.

The CI's not happy so I'll probably give up until it becomes a higher priority issue

**Comment by jberdine** [comment]
The skipruntime test is flaky, rerunning it passed.

**Comment by bennostein** [comment]
> The skipruntime test is flaky, rerunning it passed.

I suspect a race like the one Lucas hit -- working on some fixes to make them more robust to timing variability.

**Comment by mbouaziz** [comment]
Shall we merge this? (without the last commit) considering LLVM 15 and 16 seem compatible enough for our use cases.

Or do you think we should allow only one version?

File: .circleci/base.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ commands:
 jobs:
   fast-checks:
     docker:
-      - image: skiplabs/skip:latest
+      - image: skiplabs/skip:llvm16
```
Comment (beauby) [inline .circleci/base.yml:15]: Do we want to tag docker images by llvm version?

File: skiplang/compiler/src/compile.sk [hunk 2]
```diff
@@ -177,15 +177,17 @@ fun runShell(args: Array<String>, verbose: Bool = false): void {
 }
 
 fun ensureCompatibleLLVMVersion(): void {
-  kLLVMVersion = "15.";
+  kAllowedLLVMVersions = Array["15.", "16."];
```
Comment (beauby) [inline skiplang/compiler/src/compile.sk:180]: Do we want to allow multiple versions? If so, we should clarify support expectations.

File: .circleci/base.yml (hunk 1 above)
Comment (mbouaziz) [reply .circleci/base.yml:15]: If we don't plan on officially supporting and/or using more than one version, probably not.
Here it was just for testing

File: skiplang/compiler/src/compile.sk (hunk 2 above)
Comment (mbouaziz) [reply skiplang/compiler/src/compile.sk:180]: It's clearly the goal of this PR and I updated the doc accordingly.
Though, I agree I forgot to mention that we are (still) testing on LLVM 15 only.

Comment (jberdine) [reply skiplang/compiler/src/compile.sk:180]: We want to make sure that if external users try to use a version different from what we are testing, it fails, right?

**Comment by mbouaziz** [comment]
Closing in favor of what @beauby is currently doing
